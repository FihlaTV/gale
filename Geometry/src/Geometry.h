/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003, Victorian Partnership for Advanced Computing (VPAC) Ltd, 110 Victoria Street, Melbourne, 3053, Australia.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Patrick D. Sunter, Software Engineer, VPAC. (pds@vpac.org)
**	Luke J. Hodkinson, Computational Engineer, VPAC. (lhodkins@vpac.org)
**	Siew-Ching Tan, Software Engineer, VPAC. (siew@vpac.org)
**	Alan H. Lo, Computational Engineer, VPAC. (alan@vpac.org)
**	Raquibul Hassan, Computational Engineer, VPAC. (raq@vpac.org)
**
**  This library is free software; you can redistribute it and/or
**  modify it under the terms of the GNU Lesser General Public
**  License as published by the Free Software Foundation; either
**  version 2.1 of the License, or (at your option) any later version.
**
**  This library is distributed in the hope that it will be useful,
**  but WITHOUT ANY WARRANTY; without even the implied warranty of
**  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
**  Lesser General Public License for more details.
**
**  You should have received a copy of the GNU Lesser General Public
**  License along with this library; if not, write to the Free Software
**  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
**
*/
/** \file
**  Role:
**	External header file to this library.
**
** Assumptions:
**	None so far.
**
** Comments:
**	None so far.
**
** $Id: Geometry.h 4081 2007-04-27 06:20:07Z LukeHodkinson $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#ifndef __Domain_Geometry_h__
#define __Domain_Geometry_h__
	
	#include "units.h"
	#include "types.h"
	#include "isinf.h"
	#include "Dimension.h"
	#include "VectorMath.h"
	#include "TensorMath.h"
	#include "TrigMath.h"
	#include "ComplexMath.h"
	#include "Plane.h"
	#include "Edge.h"
	#include "Line.h"
	#include "RMatrix.h"
	#include "Topology.h"
	#include "IJKTopology.h"
	#include "IJK6Topology.h"
	#include "IJK26Topology.h"
	#include "IrregTopology.h"
	#include "GeometryClass.h"
	#include "BlockGeometry.h"
	#include "RefinedRegionsGeometry.h"
	#include "ShellGeometry.h"
	#include "IrregGeometry.h"
	#include "QuadEdge.h"
	#include "Delaunay.h"
	#include "ParallelDelaunay.h"
	#include "ComplexVectorMath.h"
	#include "stg_lapack.h"
	#include "FullTensorMath.h"
	#include "TensorMultMath.h"
	#include "Simplex.h"
	#include "Hex.h"
	#include "Init.h"
	#include "Finalise.h"

#endif /* __Domain_Geometry_h__ */
