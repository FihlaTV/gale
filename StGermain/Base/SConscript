import os
Import('env')

# Execute subdirectory SConscripts.
objs = SConscript('Foundation/SConscript', exports='env')
objs += SConscript('IO/SConscript', exports='env')
objs += SConscript('Container/SConscript', exports='env')
objs += SConscript('Automation/SConscript', exports='env')
objs += SConscript('Extensibility/SConscript', exports='env')
objs += SConscript('Context/SConscript', exports='env')
objs += SConscript('src/SConscript', exports='env')

# Build libraries.
libs = env.Library(env['helpers']['buildLibDir'](env), objs)
libs += env.LoadableModule(env['helpers']['buildLibDir'](env), objs)
env.Install(os.path.join(env['prefix'], 'lib'), libs)

# Build tests.
SConscript('tests/SConscript', exports='env')

# Build programs.
SConscript('FlattenXML/SConscript', exports='env')

Return('objs')
