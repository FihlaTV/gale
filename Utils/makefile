# obtain defaults for required variables according to system and project location, and then run the build.
ifndef PROJ_ROOT
	PROJ_ROOT=..
endif
include ${PROJ_ROOT}/Makefile.system

subdirs = src tests

include ${PROJ_ROOT}/Makefile.vmake
