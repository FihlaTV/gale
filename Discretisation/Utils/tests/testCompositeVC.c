/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003, Victorian Partnership for Advanced Computing (VPAC) Ltd, 110 Victoria Street, Melbourne, 3053, Australia.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Patrick D. Sunter, Software Engineer, VPAC. (pds@vpac.org)
**	Luke J. Hodkinson, Computational Engineer, VPAC. (lhodkins@vpac.org)
**	Siew-Ching Tan, Software Engineer, VPAC. (siew@vpac.org)
**	Alan H. Lo, Computational Engineer, VPAC. (alan@vpac.org)
**	Raquibul Hassan, Computational Engineer, VPAC. (raq@vpac.org)
**
**  This library is free software; you can redistribute it and/or
**  modify it under the terms of the GNU Lesser General Public
**  License as published by the Free Software Foundation; either
**  version 2.1 of the License, or (at your option) any later version.
**
**  This library is distributed in the hope that it will be useful,
**  but WITHOUT ANY WARRANTY; without even the implied warranty of
**  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
**  Lesser General Public License for more details.
**
**  You should have received a copy of the GNU Lesser General Public
**  License along with this library; if not, write to the Free Software
**  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
**
** $Id: testCompositeVC.c 4144 2007-06-13 04:33:59Z LukeHodkinson $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#include <mpi.h>
#include "Base/Base.h"

#include "Discretisation/Geometry/Geometry.h"
#include "Discretisation/Shape/Shape.h"
#include "Discretisation/Mesh/Mesh.h"
#include "Discretisation/Utils/Utils.h"

#include <stdio.h>
#include <stdlib.h>
#include <string.h>


void quadratic(Index index, Variable_Index var_I, void* context, void* result)
{
	*(double *)result = 20.0;
}


void exponential(Index index, Variable_Index var_I, void* context, void* result)
{
	*(double *)result = 30.0;
}


Mesh* buildMesh( unsigned nDims, unsigned* size, 
		     double* minCrds, double* maxCrds, 
		     ExtensionManager_Register* emReg )
{
	CartesianGenerator*	gen;
	Mesh*			mesh;
	unsigned		maxDecomp[3] = {0, 1, 1};

	gen = CartesianGenerator_New( "" );
	gen->shadowDepth = 0;
	CartesianGenerator_SetDimSize( gen, nDims );
	CartesianGenerator_SetTopologyParams( gen, size, 0, NULL, maxDecomp );
	CartesianGenerator_SetGeometryParams( gen, minCrds, maxCrds );

	mesh = Mesh_New( "" );
	Mesh_SetExtensionManagerRegister( mesh, emReg );
	Mesh_SetGenerator( mesh, gen );

	Stg_Component_Build( mesh, NULL, False );
	Stg_Component_Initialise( mesh, NULL, False );

	KillObject( mesh->generator );

	return mesh;
}


int main(int argc, char *argv[])
{
	MPI_Comm		CommWorld;
	int		rank;
	int		procCount;
	int		procToWatch;
	Stream*		stream;
	
	Dictionary*		dictionary;
	XML_IO_Handler*		io_handler;
	
	unsigned	nDims = 3;
	unsigned	meshSize[3] = {3, 3, 3};
	double		minCrds[3] = {0.0, 0.0, 0.0};
	double		maxCrds[3] = {1.0, 1.0, 1.0};
	Mesh*		mesh;
	
	Variable*			var[7];
	Variable_Register*		variable_Register;
	WallVC*				vc;
	CompositeVC*			cvc;
	ConditionFunction*		quadCF;
	ConditionFunction*		expCF;
	ConditionFunction_Register*	conFunc_Register;
	ExtensionManager_Register*		extensionMgr_Register;
	
	double*		array[7];
	char*		vcKey[] = {"WallVC_Front", "WallVC_Back", "WallVC_Left", "WallVC_Right",
				"WallVC_Top", "WallVC_Bottom"};
	char*		vcKeyName[] = {"WallVC_FrontName", "WallVC_BackName", "WallVC_LeftName", "WallVC_RightName",
				"WallVC_TopName", "WallVC_BottomName"};
	char*		varName[] = {"x", "y", "z", "vx", "vy", "vz", "temp"};

	unsigned	nDomains;
	
	Index	i, j, k;
	
	/* Initialise MPI, get world info */
	MPI_Init(&argc, &argv);
	MPI_Comm_dup( MPI_COMM_WORLD, &CommWorld );
	MPI_Comm_size(CommWorld, &procCount);
	MPI_Comm_rank(CommWorld, &rank);
	
	Base_Init( &argc, &argv );
	
	DiscretisationGeometry_Init( &argc, &argv );
	DiscretisationShape_Init( &argc, &argv );
	DiscretisationMesh_Init( &argc, &argv );
	DiscretisationUtils_Init( &argc, &argv );
	MPI_Barrier( CommWorld ); /* Ensures copyright info always come first in output */

	Journal_Enable_NamedStream( Info_Type, CartesianGenerator_Type, False );

	io_handler = XML_IO_Handler_New();
	
	stream = Journal_Register (Info_Type, "myStream");

	procToWatch = argc >= 2 ? atoi(argv[1]) : 0;
	
	dictionary = Dictionary_New();
	Dictionary_Add(dictionary, "outputPath", Dictionary_Entry_Value_FromString("./output"));
	IO_Handler_ReadAllFromFile(io_handler, "data/wallVC.xml", dictionary);

	extensionMgr_Register = ExtensionManager_Register_New();
	
	/* Create a mesh. */
	mesh = buildMesh( nDims, meshSize, minCrds, maxCrds, extensionMgr_Register );
	nDomains = Mesh_GetDomainSize( mesh, MT_VERTEX );
	
	/* Create CF stuff */
	quadCF = ConditionFunction_New(quadratic, "quadratic");
	expCF = ConditionFunction_New(exponential, "exponential");
	conFunc_Register = ConditionFunction_Register_New();
	ConditionFunction_Register_Add(conFunc_Register, quadCF);
	ConditionFunction_Register_Add(conFunc_Register, expCF);
	
	/* Create variable register */
	variable_Register = Variable_Register_New();
	
	/* Create variables */
	for (i = 0; i < 6; i++) {
		array[i] = Memory_Alloc_Array( double, nDomains, "array[i]" );
		var[i] = Variable_NewScalar( varName[i], Variable_DataType_Double, &nDomains, (void**)&array[i], 0 ); 
		Variable_Register_Add(variable_Register, var[i]);
	}
	array[6] = Memory_Alloc_Array( double, nDomains*5, "array[6]" );
	var[6] = Variable_NewVector( varName[6], Variable_DataType_Double, 5, &nDomains, (void**)&array[6], 0 );
	Variable_Register_Add(variable_Register, var[6]);
	Variable_Register_BuildAll(variable_Register);
	
	/* Create CompositeVC */
	cvc = CompositeVC_New( "CompositeVC", variable_Register, conFunc_Register, dictionary, mesh );
	
	for (i = 0; i < 6; i++)
	{
		vc = WallVC_New( vcKeyName[i], vcKey[i], variable_Register, conFunc_Register, dictionary, mesh );
		Stg_Component_Build( vc, 0, False );
		CompositeVC_Add(cvc, vc, True);
	}
	
	Stg_Component_Build( cvc, 0, False );
	
	for (j = 0; j < 6; j++)
		memset(array[j], 0, sizeof(double)*nDomains);
	memset(array[6], 0, sizeof(double)*nDomains*5);
	VariableCondition_Apply(cvc, NULL);
	
	if (rank == procToWatch)
	{
		Stg_Class_Print(cvc, stream);
		printf("\n");
		for (j = 0; j < 6; j++)
		{
			printf("\nvar[%u]: %.2lf", j, array[j][0]);
			for (k = 1; k < nDomains; k++)
				printf(", %.2lf", array[j][k]);
		}
		printf("\nvar[7]: %.2lf", array[6][0]);
		for (j = 1; j < nDomains*5; j++)
			printf(", %.2lf", array[6][j]);
		printf("\n\n");
			
		for (j = 0; j < 7; j++)
		{
			for (k = 0; k < nDomains; k++)
				printf("%s ", VariableCondition_IsCondition(cvc, k, j) ? "True " : "False");
			printf("\n");
		}
		printf("\n");
			
		for (j = 0; j < 7; j++)
		{
			for (k = 0; k < nDomains; k++)
			{
				VariableCondition_ValueIndex	valIndex;
				
				valIndex = VariableCondition_GetValueIndex(cvc, k, j);
				if (valIndex != (unsigned)-1)
					printf("%03u ", valIndex);
				else
					printf("XXX ");
			}
			printf("\n");
		}
		printf("\n");
	}

	Stg_Class_Delete(cvc);
	Stg_Class_Delete(variable_Register);
	for (i = 0; i < 7; i++)
	{
		Stg_Class_Delete(var[i]);
		if (array[i]) Memory_Free(array[i]);
	}
	Stg_Class_Delete(conFunc_Register);
	Stg_Class_Delete(quadCF);
	Stg_Class_Delete(expCF);
	Stg_Class_Delete(dictionary);
	FreeObject( mesh );
	
	DiscretisationUtils_Finalise();
	DiscretisationMesh_Finalise();
	DiscretisationShape_Finalise();
	DiscretisationGeometry_Finalise();
	
	Base_Finalise();
	
	/* Close off MPI */
	MPI_Finalize();
	
	return 0; /* success */
}
