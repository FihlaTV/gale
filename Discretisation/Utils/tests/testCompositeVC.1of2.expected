StGermain Framework revision 3842. Copyright (C) 2003-2005 VPAC.
CompositeVC (ptr): 0x112bf88
	dictionary (ptr): 0x111df68
	itemCount: 6
	itemTbl (ptr): 0x112c298
		itemTbl[0] (ptr): 0x112c2f8
		itemTbl[1] (ptr): 0x112cda8
		itemTbl[2] (ptr): 0x112d228
		itemTbl[3] (ptr): 0x112d618
		itemTbl[4] (ptr): 0x112dc78
		itemTbl[5] (ptr): 0x112e108
	iOwnTbl (ptr): 0x112c2c8
		iOwnTbl[0]: True
		iOwnTbl[1]: True
		iOwnTbl[2]: True
		iOwnTbl[3]: True
		iOwnTbl[4]: True
		iOwnTbl[5]: True
	_size: 8
	_delta: 8
VariableCondition (ptr): 0x112bf88
Stg_Component (ptr): 0x112bf88
Stg_Object (ptr): 0x112bf88
	Stg_Class (ptr): 0x112bf88
		sizeOfSelf: 204
		_deleteSelf: Yes
		type: CompositeVC
		_delete (func ptr): 0x571970
		_print (func ptr): 0x571a78
		_copy (func ptr): 0x571c80
	name: CompositeVC
	nameAllocationType: NON_GLOBAL
	_defaultConstructor (func ptr): 0x571088
	_construct (func ptr): 0x571778
	_build (func ptr): 0x572090
	_initialise (func ptr): 0x56e31c
	_execute (func ptr): 0x56e410
	_destroy (func ptr): 0x56e454
	isConstructed: True
	isBuilt: True
	isInitialised: False
	hasExecuted: False
	isDestroyed: False
	constructor function name: CompositeVC-Construct
	build function name: CompositeVC-Build
	initialise function name: CompositeVC-Initialise
	execute function name: CompositeVC-Execute
	destroy function name: CompositeVC-Destroy
	_getSet (func ptr): 0x572250
	_getVariableCount (func ptr): 0x5723ac
	_getVariableIndex (func ptr): 0x5725d4
	_getValueIndex (func ptr): 0x572838
	_getValueCount (func ptr): 0x572b2c
	_getValue (func ptr): 0x572bf0
	variable_Register (ptr): 0x111ec28
	conFunc_Register (ptr): 0x111ed18
	_set (ptr): 0x112e6b8
	indexCount: 28
	indexTbl (ptr): 0x112e898
		indexTbl[0]: 0
		indexTbl[1]: 1
		indexTbl[2]: 2
		indexTbl[3]: 3
		indexTbl[4]: 4
		indexTbl[5]: 5
		indexTbl[6]: 6
		indexTbl[7]: 7
		indexTbl[8]: 8
		indexTbl[9]: 9
		indexTbl[10]: 10
		indexTbl[11]: 11
		indexTbl[12]: 12
		indexTbl[13]: 13
		indexTbl[14]: 14
		indexTbl[15]: 15
		indexTbl[16]: 16
		indexTbl[17]: 17
		indexTbl[18]: 18
		indexTbl[19]: 19
		indexTbl[20]: 20
		indexTbl[21]: 23
		indexTbl[22]: 24
		indexTbl[23]: 27
		indexTbl[24]: 28
		indexTbl[25]: 29
		indexTbl[26]: 30
		indexTbl[27]: 31
	vcVarCountTbl (ptr): 0x112e918
		vcVarCountTbl[0]: 4
		vcVarCountTbl[1]: 4
		vcVarCountTbl[2]: 4
		vcVarCountTbl[3]: 4
		vcVarCountTbl[4]: 3
		vcVarCountTbl[5]: 3
		vcVarCountTbl[6]: 3
		vcVarCountTbl[7]: 4
		vcVarCountTbl[8]: 3
		vcVarCountTbl[9]: 3
		vcVarCountTbl[10]: 3
		vcVarCountTbl[11]: 4
		vcVarCountTbl[12]: 3
		vcVarCountTbl[13]: 3
		vcVarCountTbl[14]: 3
		vcVarCountTbl[15]: 4
		vcVarCountTbl[16]: 4
		vcVarCountTbl[17]: 1
		vcVarCountTbl[18]: 1
		vcVarCountTbl[19]: 1
		vcVarCountTbl[20]: 3
		vcVarCountTbl[21]: 1
		vcVarCountTbl[22]: 3
		vcVarCountTbl[23]: 1
		vcVarCountTbl[24]: 3
		vcVarCountTbl[25]: 3
		vcVarCountTbl[26]: 3
		vcVarCountTbl[27]: 4
	vcTbl (ptr): 0x3016608
		vcTbl[0][0]:
			varIndex: 6
			valIndex: 3
		vcTbl[0][1]:
			varIndex: 3
			valIndex: 11
		vcTbl[0][2]:
			varIndex: 4
			valIndex: 12
		vcTbl[0][3]:
			varIndex: 5
			valIndex: 13
		vcTbl[1][0]:
			varIndex: 6
			valIndex: 3
		vcTbl[1][1]:
			varIndex: 3
			valIndex: 7
		vcTbl[1][2]:
			varIndex: 4
			valIndex: 8
		vcTbl[1][3]:
			varIndex: 5
			valIndex: 9
		vcTbl[2][0]:
			varIndex: 6
			valIndex: 3
		vcTbl[2][1]:
			varIndex: 3
			valIndex: 7
		vcTbl[2][2]:
			varIndex: 4
			valIndex: 8
		vcTbl[2][3]:
			varIndex: 5
			valIndex: 9
		vcTbl[3][0]:
			varIndex: 6
			valIndex: 10
		vcTbl[3][1]:
			varIndex: 3
			valIndex: 7
		vcTbl[3][2]:
			varIndex: 4
			valIndex: 8
		vcTbl[3][3]:
			varIndex: 5
			valIndex: 9
		vcTbl[4][0]:
			varIndex: 3
			valIndex: 11
		vcTbl[4][1]:
			varIndex: 4
			valIndex: 12
		vcTbl[4][2]:
			varIndex: 5
			valIndex: 13
		vcTbl[5][0]:
			varIndex: 3
			valIndex: 7
		vcTbl[5][1]:
			varIndex: 4
			valIndex: 8
		vcTbl[5][2]:
			varIndex: 5
			valIndex: 9
		vcTbl[6][0]:
			varIndex: 3
			valIndex: 7
		vcTbl[6][1]:
			varIndex: 4
			valIndex: 8
		vcTbl[6][2]:
			varIndex: 5
			valIndex: 9
		vcTbl[7][0]:
			varIndex: 3
			valIndex: 7
		vcTbl[7][1]:
			varIndex: 4
			valIndex: 8
		vcTbl[7][2]:
			varIndex: 5
			valIndex: 9
		vcTbl[7][3]:
			varIndex: 6
			valIndex: 10
		vcTbl[8][0]:
			varIndex: 3
			valIndex: 11
		vcTbl[8][1]:
			varIndex: 4
			valIndex: 12
		vcTbl[8][2]:
			varIndex: 5
			valIndex: 13
		vcTbl[9][0]:
			varIndex: 3
			valIndex: 7
		vcTbl[9][1]:
			varIndex: 4
			valIndex: 8
		vcTbl[9][2]:
			varIndex: 5
			valIndex: 9
		vcTbl[10][0]:
			varIndex: 3
			valIndex: 7
		vcTbl[10][1]:
			varIndex: 4
			valIndex: 8
		vcTbl[10][2]:
			varIndex: 5
			valIndex: 9
		vcTbl[11][0]:
			varIndex: 3
			valIndex: 7
		vcTbl[11][1]:
			varIndex: 4
			valIndex: 8
		vcTbl[11][2]:
			varIndex: 5
			valIndex: 9
		vcTbl[11][3]:
			varIndex: 6
			valIndex: 10
		vcTbl[12][0]:
			varIndex: 3
			valIndex: 11
		vcTbl[12][1]:
			varIndex: 4
			valIndex: 12
		vcTbl[12][2]:
			varIndex: 5
			valIndex: 13
		vcTbl[13][0]:
			varIndex: 3
			valIndex: 7
		vcTbl[13][1]:
			varIndex: 4
			valIndex: 8
		vcTbl[13][2]:
			varIndex: 5
			valIndex: 9
		vcTbl[14][0]:
			varIndex: 3
			valIndex: 7
		vcTbl[14][1]:
			varIndex: 4
			valIndex: 8
		vcTbl[14][2]:
			varIndex: 5
			valIndex: 9
		vcTbl[15][0]:
			varIndex: 3
			valIndex: 7
		vcTbl[15][1]:
			varIndex: 4
			valIndex: 8
		vcTbl[15][2]:
			varIndex: 5
			valIndex: 9
		vcTbl[15][3]:
			varIndex: 6
			valIndex: 10
		vcTbl[16][0]:
			varIndex: 6
			valIndex: 3
		vcTbl[16][1]:
			varIndex: 3
			valIndex: 11
		vcTbl[16][2]:
			varIndex: 4
			valIndex: 12
		vcTbl[16][3]:
			varIndex: 5
			valIndex: 13
		vcTbl[17][0]:
			varIndex: 6
			valIndex: 3
		vcTbl[18][0]:
			varIndex: 6
			valIndex: 3
		vcTbl[19][0]:
			varIndex: 6
			valIndex: 10
		vcTbl[20][0]:
			varIndex: 3
			valIndex: 11
		vcTbl[20][1]:
			varIndex: 4
			valIndex: 12
		vcTbl[20][2]:
			varIndex: 5
			valIndex: 13
		vcTbl[21][0]:
			varIndex: 6
			valIndex: 10
		vcTbl[22][0]:
			varIndex: 3
			valIndex: 11
		vcTbl[22][1]:
			varIndex: 4
			valIndex: 12
		vcTbl[22][2]:
			varIndex: 5
			valIndex: 13
		vcTbl[23][0]:
			varIndex: 6
			valIndex: 10
		vcTbl[24][0]:
			varIndex: 3
			valIndex: 11
		vcTbl[24][1]:
			varIndex: 4
			valIndex: 12
		vcTbl[24][2]:
			varIndex: 5
			valIndex: 13
		vcTbl[25][0]:
			varIndex: 3
			valIndex: 0
		vcTbl[25][1]:
			varIndex: 4
			valIndex: 1
		vcTbl[25][2]:
			varIndex: 5
			valIndex: 2
		vcTbl[26][0]:
			varIndex: 3
			valIndex: 0
		vcTbl[26][1]:
			varIndex: 4
			valIndex: 1
		vcTbl[26][2]:
			varIndex: 5
			valIndex: 2
		vcTbl[27][0]:
			varIndex: 3
			valIndex: 0
		vcTbl[27][1]:
			varIndex: 4
			valIndex: 1
		vcTbl[27][2]:
			varIndex: 5
			valIndex: 2
		vcTbl[27][3]:
			varIndex: 6
			valIndex: 10
	valueCount: 14
	valueTbl (ptr): 0x112e9d8
		valueTbl[0]:
			type: VC_ValueType_Double
			asDouble: 2
		valueTbl[1]:
			type: VC_ValueType_Double
			asDouble: 3
		valueTbl[2]:
			type: VC_ValueType_Double
			asDouble: 1
		valueTbl[3]:
			type: VC_ValueType_DoubleArray
			arraySize: 5
			asDoubleArray (ptr): 0x112d0c8
				asDoubleArray[0]: 5
				asDoubleArray[1]: 6
				asDoubleArray[2]: 7
				asDoubleArray[3]: 8
				asDoubleArray[4]: 9
		valueTbl[4]:
			type: VC_ValueType_CFIndex
			asCFIndex: 0
		valueTbl[5]:
			type: VC_ValueType_CFIndex
			asCFIndex: 0
		valueTbl[6]:
			type: VC_ValueType_CFIndex
			asCFIndex: 0
		valueTbl[7]:
			type: VC_ValueType_CFIndex
			asCFIndex: 1
		valueTbl[8]:
			type: VC_ValueType_CFIndex
			asCFIndex: 1
		valueTbl[9]:
			type: VC_ValueType_CFIndex
			asCFIndex: 1
		valueTbl[10]:
			type: VC_ValueType_DoubleArray
			arraySize: 5
			asDoubleArray (ptr): 0x112dfb8
				asDoubleArray[0]: 5
				asDoubleArray[1]: 6
				asDoubleArray[2]: 7
				asDoubleArray[3]: 8
				asDoubleArray[4]: 9
		valueTbl[11]:
			type: VC_ValueType_Double
			asDouble: 2
		valueTbl[12]:
			type: VC_ValueType_Double
			asDouble: 3
		valueTbl[13]:
			type: VC_ValueType_Double
			asDouble: 1


var[0]: 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00
var[1]: 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00
var[2]: 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00
var[3]: 2.00, 30.00, 30.00, 30.00, 2.00, 30.00, 30.00, 30.00, 2.00, 30.00, 30.00, 30.00, 2.00, 30.00, 30.00, 30.00, 2.00, 0.00, 0.00, 0.00, 2.00, 0.00, 0.00, 0.00, 2.00, 0.00, 0.00, 0.00, 2.00, 2.00, 2.00, 2.00
var[4]: 3.00, 30.00, 30.00, 30.00, 3.00, 30.00, 30.00, 30.00, 3.00, 30.00, 30.00, 30.00, 3.00, 30.00, 30.00, 30.00, 3.00, 0.00, 0.00, 0.00, 3.00, 0.00, 0.00, 0.00, 3.00, 0.00, 0.00, 0.00, 3.00, 3.00, 3.00, 3.00
var[5]: 1.00, 30.00, 30.00, 30.00, 1.00, 30.00, 30.00, 30.00, 1.00, 30.00, 30.00, 30.00, 1.00, 30.00, 30.00, 30.00, 1.00, 0.00, 0.00, 0.00, 1.00, 0.00, 0.00, 0.00, 1.00, 0.00, 0.00, 0.00, 1.00, 1.00, 1.00, 1.00
var[7]: 5.00, 6.00, 7.00, 8.00, 9.00, 5.00, 6.00, 7.00, 8.00, 9.00, 5.00, 6.00, 7.00, 8.00, 9.00, 5.00, 6.00, 7.00, 8.00, 9.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 5.00, 6.00, 7.00, 8.00, 9.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 5.00, 6.00, 7.00, 8.00, 9.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 5.00, 6.00, 7.00, 8.00, 9.00, 5.00, 6.00, 7.00, 8.00, 9.00, 5.00, 6.00, 7.00, 8.00, 9.00, 5.00, 6.00, 7.00, 8.00, 9.00, 5.00, 6.00, 7.00, 8.00, 9.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 5.00, 6.00, 7.00, 8.00, 9.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 5.00, 6.00, 7.00, 8.00, 9.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 5.00, 6.00, 7.00, 8.00, 9.00

False False False False False False False False False False False False False False False False False False False False False False False False False False False False False False False False 
False False False False False False False False False False False False False False False False False False False False False False False False False False False False False False False False 
False False False False False False False False False False False False False False False False False False False False False False False False False False False False False False False False 
True  True  True  True  True  True  True  True  True  True  True  True  True  True  True  True  True  False False False True  False False False True  False False False True  True  True  True  
True  True  True  True  True  True  True  True  True  True  True  True  True  True  True  True  True  False False False True  False False False True  False False False True  True  True  True  
True  True  True  True  True  True  True  True  True  True  True  True  True  True  True  True  True  False False False True  False False False True  False False False True  True  True  True  
True  True  True  True  False False False True  False False False True  False False False True  True  True  True  True  False False False True  False False False True  False False False True  

XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX 
XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX 
XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX XXX 
011 007 007 007 011 007 007 007 011 007 007 007 011 007 007 007 011 XXX XXX XXX 011 XXX XXX XXX 011 XXX XXX XXX 011 000 000 000 
012 008 008 008 012 008 008 008 012 008 008 008 012 008 008 008 012 XXX XXX XXX 012 XXX XXX XXX 012 XXX XXX XXX 012 001 001 001 
013 009 009 009 013 009 009 009 013 009 009 009 013 009 009 009 013 XXX XXX XXX 013 XXX XXX XXX 013 XXX XXX XXX 013 002 002 002 
003 003 003 010 XXX XXX XXX 010 XXX XXX XXX 010 XXX XXX XXX 010 003 003 003 010 XXX XXX XXX 010 XXX XXX XXX 010 XXX XXX XXX 010 

